sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/ComponentOptions.scala

```diff
@@ -14,31 +14,25 @@
  * limitations under the License.
  */
 
-package com.akkaserverless.javasdk;
+package com.akkaserverless.javasdk.impl
 
-import com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntityOptions;
-import com.akkaserverless.javasdk.impl.eventsourcedentity.EventSourcedEntityOptionsImpl;
+import com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntityOptions
+import com.akkaserverless.javasdk.impl.eventsourcedentity.EventSourcedEntityOptionsImpl
 
 import java.util.Collections;
 import java.util.Set;
 
-public interface ComponentOptions {
+trait ComponentOptions {
 
   /**
-   * @return the headers requested to be forwarded as metadata (cannot be mutated, use
-   *     withForwardHeaders)
+   * @return
+   *   the headers requested to be forwarded as metadata (cannot be mutated, use withForwardHeaders)
    */
-  Set<String> forwardHeaders();
+  def forwardHeaders(): Set[String]
 
   /**
-   * Ask Akka Serverless to forward these headers from the incoming request as metadata headers for
-   * the incoming commands. By default no headers except "X-Server-Timing" are forwarded.
+   * Ask Akka Serverless to forward these headers from the incoming request as metadata headers for the incoming
+   * commands. By default no headers except "X-Server-Timing" are forwarded.
    */
-  ComponentOptions withForwardHeaders(Set<String> headers);
-
-  /** Create a default components option */
-  static ComponentOptions defaults() {
-    return new EventSourcedEntityOptionsImpl(
-        0, PassivationStrategy.defaultTimeout(), Collections.emptySet());
-  }
+  def withForwardHeaders(headers: Set[String]): ComponentOptions
```

Comment:
^ regarding `Set`

Reply:
This is public api and it's somewhat hidden (strange) to only have it via `impl`. We should add override in all that `extends ComponentOptions`, for example ActionOptions

Reply:
ok